x-pack/plugins/infra/public/components/asset_details/tabs/overview/metadata_summary/metadata_summary_list.tsx

```diff
@@ -78,54 +80,56 @@ const MetadataSummaryListWrapper = ({
     showTab(ContentTabIds.METADATA);
   };
 
+  const ExtraActions = () => (
+    <EuiButtonEmpty
+      data-test-subj="infraAssetDetailsMetadataShowAllButton"
+      onClick={onClick}
+      size="xs"
+      flush="both"
+      iconSide="right"
+      iconType="sortRight"
+      key="metadata-link"
+    >
+      <FormattedMessage
+        id="xpack.infra.assetDetailsEmbeddable.metadataSummary.showAllMetadataButton"
+        defaultMessage="Show all"
+      />
+    </EuiButtonEmpty>
+  );
+
   return (
     <EuiFlexGroup gutterSize="s" justifyContent="spaceBetween" direction="column" wrap>
-      <EuiFlexGroup direction="column" gutterSize="xs">
-        <EuiFlexGroup justifyContent="spaceBetween" responsive={false}>
-          <EuiFlexItem grow={false}>
-            <MetadataSectionTitle />
-          </EuiFlexItem>
-          <EuiFlexItem grow={false} key="metadata-link">
-            <EuiButtonEmpty
-              data-test-subj="infraAssetDetailsMetadataShowAllButton"
-              onClick={onClick}
-              size="xs"
-              flush="both"
-              iconSide="right"
-              iconType="sortRight"
-            >
-              <FormattedMessage
-                id="xpack.infra.assetDetailsEmbeddable.metadataSummary.showAllMetadataButton"
-                defaultMessage="Show all"
-              />
-            </EuiButtonEmpty>
-          </EuiFlexItem>
-        </EuiFlexGroup>
-        <EuiFlexGroup>
-          <EuiFlexItem>
-            <MetadataExplanationMessage />
-          </EuiFlexItem>
-        </EuiFlexGroup>
-      </EuiFlexGroup>
-      <EuiFlexGroup>
-        {visibleMetadata.map(
-          (metadataValue) =>
-            metadataValue && (
-              <EuiFlexItem key={metadataValue.field} grow={false}>
-                <EuiDescriptionList data-test-subj="infraMetadataSummaryItem" compressed>
-                  <MetadataHeader metadataValue={metadataValue} />
-                  <EuiDescriptionListDescription>
-                    {metadataLoading && !metadataValue.value ? (
-                      <EuiLoadingSpinner />
-                    ) : (
-                      <ExpandableContent values={metadataValue.value ?? NOT_AVAILABLE_LABEL} />
-                    )}
-                  </EuiDescriptionListDescription>
-                </EuiDescriptionList>
-              </EuiFlexItem>
-            )
-        )}
-      </EuiFlexGroup>
+      <CollapsibleSection
+        title={MetadataSectionTitle}
+        shouldCollapse={true}
+        data-test-subj="infraAssetDetailsMetadataCollapsible"
+        id="metadata"
+        extraAction={<ExtraActions />}
+      >
+        <>
+          <MetadataExplanationMessage />
+          <EuiSpacer size="s" />
+          <EuiFlexGroup>
+            {visibleMetadata.map(
+              (metadataValue) =>
+                metadataValue && (
```

Comment:
I could filter out the undefined values first and them only render the ones defined.
```suggestion
            {visibleMetadata.filter(metadataValue => metadataValue).map(
              (metadataValue) =>
```